compressor/tests/test_storages.py

```diff
@@ -87,5 +101,35 @@ def test_offline_manifest_storage(self):
         self.assertTrue(
             os.path.exists(os.path.join(settings.COMPRESS_ROOT, "CACHE", "test.txt"))
         )
-        # Check that the file is stored at the same default location as before the new manifest storage.
+        # Check that the file is stored at the same default location as before
+        # the new manifest storage.
         self.assertTrue(self.default_storage.exists(os.path.join("CACHE", "test.txt")))
+
+
+class CompressorFileNameTestCase(TestCase):
+    @override_settings(
+        COMPRESS_ENABLED=True,
+        DEBUG=False,
+        COMPRESS_STORAGE=(
+            "compressor.tests.test_storages.DummyPathNotImplementedStorage"
+        ),
+    )
+    def test_storage_without_path_fallback(self):
+        """
+        Remote storages not implementing path need a fallback to a private
+        instance of CompressorFileStorage. This must not be dependent on
+        project settings.
+        """
+        old_default_storage = storage.default_storage
+        storage.default_storage = storage.DefaultStorage()
```

Comment:
get_filename uses the default storage, so we need to clear (and reset) in the test case.